test/sql/test_decimal/R/test_decimal_overflow

```diff
@@ -0,0 +1,123 @@
+-- name: test_decimal_overflow
+CREATE TABLE `t_decimal_overflow` (
+  `c_id` int(11) NOT NULL,
+  `c_d32` decimal32(9,3) NOT NULL,
+  `c_d64` decimal64(18,5) NOT NULL,
+  `c_d128` decimal128(38,7) NOT NULL
+) ENGINE=OLAP
+DUPLICATE KEY(`c_id`)
+DISTRIBUTED BY HASH(`c_id`) BUCKETS 10
+PROPERTIES (
+ "replication_num" = "1"
+);
+-- result:
+-- !result
+INSERT INTO `t_decimal_overflow` (c_id, c_d32, c_d64, c_d128) values
+   (1, 999999.99, 9999999999999.99999, 9999999999999999999999999999999.9999999),
+   (2, -999999.99, -9999999999999.99999, -9999999999999999999999999999999.9999999);
+-- result:
+-- !result
+select 274.97790000000000000000 * (round(1103.00000000000000000000 * 1.0000,16) /round(1103.00000000000000000000,16));
+-- result:
+None
+-- !result
+select /*+ SET_VAR(sql_mode='ERROR_IF_OVERFLOW')*/ 274.97790000000000000000 * (round(1103.00000000000000000000 * 1.0000,16) /round(1103.00000000000000000000,16));
+-- result:
+E: (1064, 'Expr evaluate meet error: Decimal overflow')
+-- !result
+select cast(c_d32 * c_d32 as decimal32) from t_decimal_overflow where c_id = 1;
+-- result:
+None
+-- !result
+select cast(c_d32 * c_d32 as decimal32) from t_decimal_overflow where c_id = 2;
+-- result:
+None
+-- !result
+select cast(c_d64 * c_d64 as decimal64) from t_decimal_overflow where c_id = 1;
+-- result:
+None
+-- !result
+select cast(c_d64 * c_d64 as decimal64) from t_decimal_overflow where c_id = 2;
+-- result:
+None
+-- !result
+select cast(c_d128 * c_d128 as decimal128) from t_decimal_overflow where c_id = 1;
+-- result:
+None
+-- !result
+select cast(c_d128 * c_d128 as decimal128) from t_decimal_overflow where c_id = 2;
+-- result:
+None
+-- !result
+select cast(c_d32 * 1.000 as decimal32) from t_decimal_overflow where c_id = 1;
+-- result:
+None
+-- !result
+select cast(c_d32 * 1.000 as decimal32) from t_decimal_overflow where c_id = 2;
+-- result:
+None
+-- !result
+select cast(c_d64 * 1.000000 as decimal64) from t_decimal_overflow where c_id = 1;
+-- result:
+None
+-- !result
+select cast(c_d64 * 1.000000 as decimal64) from t_decimal_overflow where c_id = 2;
+-- result:
+None
+-- !result
+select cast(c_d128 * 1.000000000 as decimal128) from t_decimal_overflow where c_id = 1;
+-- result:
+None
+-- !result
+select cast(c_d128 * 1.000000000 as decimal128) from t_decimal_overflow where c_id = 2;
+-- result:
+None
+-- !result
+select /*+ SET_VAR(sql_mode='ERROR_IF_OVERFLOW')*/ cast(c_d32 * c_d32 as decimal32) from t_decimal_overflow where c_id = 1;
+-- result:
+E: (1064, 'Expr evaluate meet error: Decimal overflow')
```

Comment:
from the error, we can't know which expr report this error, can the error specify the operator?

Reply:
Add more precise information.